x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -19,25 +21,65 @@ export const hostMiddlewareFactory: ImmutableMiddlewareFactory<HostState> = (cor
       hasSelectedHost(state) !== true
     ) {
       const { page_index: pageIndex, page_size: pageSize } = uiQueryParams(state);
+      let hostResponse;
+
       try {
-        const response = await coreStart.http.post<HostResultList>('/api/endpoint/metadata', {
+        hostResponse = await coreStart.http.post<HostResultList>('/api/endpoint/metadata', {
           body: JSON.stringify({
             paging_properties: [{ page_index: pageIndex }, { page_size: pageSize }],
           }),
         });
-        response.request_page_index = Number(pageIndex);
+        hostResponse.request_page_index = Number(pageIndex);
+
         dispatch({
           type: 'serverReturnedHostList',
-          payload: response,
+          payload: hostResponse,
         });
       } catch (error) {
         dispatch({
           type: 'serverFailedToReturnHostList',
           payload: error,
         });
       }
+
+      // No hosts, so we should check to see if there are policies for onboarding
+      if (hostResponse && hostResponse.hosts.length === 0) {
+        const http = coreStart.http;
```

Comment:
maybe this should be defined at top - below line 17 - and used trough out this module